Sparkle/SUBinaryDeltaCreate.m

```diff
@@ -138,11 +220,25 @@ int createBinaryDelta(NSString *source, NSString *destination, NSString *patchFi
         }
 
         NSDictionary *info = infoForFile(ent);
+        if (!info) {
+            fprintf(stderr, "Failed to retrieve info for file %s", ent->fts_path);
+            return 1;
+        }
         originalTreeState[key] = info;
+        
+        if (aclExists(ent)) {
+            fprintf(stderr, "Diffing ACLs are not supported. Detected ACL in before-tree on file %s", ent->fts_path);
+            return 1;
+        }
     }
     fts_close(fts);
+    
+    NSString *beforeHash = hashOfTreeWithVersion(source, majorVersion);
```

Comment:
I think this could be complicated. Let's take the scenario where you create a patch between two trees that both contain the same file. After creating the patch, you then alter the permissions of the file in the original tree and we're assuming this doesn't change the directory hash. When applying the patch to the original tree, BinaryDelta has no way to know to update the permissions of that file and we'll end up with an after-tree with the wrong permissions. We would have to do something like storing the permissions of every file in the diff that is not set to be extracted from the xar, and we wouldn't be able to mark any file as "not needed to patch" early on.